File: server/withssafy/src/main/java/com/ssafy/withssafy/dto/board/BoardRequest.java
package com.ssafy.withssafy.dto.board;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

@Data
public class BoardRequest {
    @JsonIgnore
    private Long id;
    private Long userId;
    private Long typeId;
    private String content;
    private String photoPath;
    private String title;
    private String writeDateTime;

    public void setWriteDateTime(){
        this.writeDateTime = String.valueOf(System.currentTimeMillis());
    }
}

File: application/WithSSAFY/app/src/main/java/com/ssafy/withssafy/src/dto/board/BoardRequest.kt
package com.ssafy.withssafy.src.dto.board

data class BoardRequest(
    val id: Int,
    val typeId: Int,
    val userId: Int,
    val title: String,
    val content: String,
    val photoPath: String,
    val writeDateTime: String?
) {
    // insert
    constructor(typeId: Int, userId: Int, title: String, content: String, photoPath: String) : this(id = 0, typeId = typeId, userId = userId, title = title, content = content, photoPath = photoPath, null)

    // update
    constructor(id: Int, typeId: Int, userId: Int, title: String, content: String, photoPath: String) : this(id = id, typeId = typeId, userId = userId, title = title, content = content, photoPath = photoPath, null)
}